walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,251 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.TreeSet;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  /**
+   * Finds the possible TimeRanges where the mandatory attendees can attend.
+   * @param eventsList sorted list of events in ascending start time order
+   * @param request includes mandatory attendees and duration of meeting
+   * @return collection of possible TimeRanges the meeting can be scheduled in  
```

Comment:
```suggestion
   * Finds the possible TimeRanges where the mandatory attendees can attend.
   *
   * @param eventsList sorted list of events in ascending start time order
   * @param request includes mandatory attendees and duration of meeting
   * @return collection of possible TimeRanges the meeting can be scheduled in  
```

Some for the other similar places.

More examples: https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html